lib/posix-process/clone.c

```diff
@@ -388,6 +388,25 @@ static int _clone(struct clone_args *cl_args, size_t cl_args_len,
 	uk_pr_debug(")\n");
 #endif /* UK_DEBUG */
 
+	/* CLONE_VM requires that the child operates on the same memory
+	* space as the parent.
+	*
+	* Assign the parent's stack. We assign the user TLS to the
+	* parent's in clone_setup_arch_sysregs(). We copy the registers
+	* later below.
+	*/
+	if ((flags & CLONE_VM)) {
+		if (!cl_args->stack && !cl_args->stack_size) {
+			uk_pr_debug("Using parent's sp @ 0x%lx\n", execenv->regs.rsp);
+			cl_args->stack = execenv->regs.rsp; /* FIXME */
```

Comment:
I assume FIXME cus x86 specific, right?

Reply:
Correct. I now included your patch with `ukarch_regs_get_sp()` to resolve this.